### PR TITLE
Replacing monitor into TidbMonitor in chart

### LINKED ISSUE #1558: Use TidbMonitor for tidbcluster if monitor enabled
## Feature Request

As we already support `TidbMonitor`, we should use `TidbMonitor` instead of `monitor template` if `monitor` is enabled in `values.yaml` in `tidb-cluster` helm chart.

### PR DESCRIPTION
**Pull request by longfangsong**
### What problem does this PR solve? 

try fix #1558

Also fix the "Release Notes Language Style Guide" url in pull_request_template.md

### What is changed and how does it work?

Use a single `monitor.yaml` which deploys a `TidbMonitor` instead of `monitor-*.yaml` which deploy lot of different components to make monitor work.

### Check List 

Tests 

 - No code

Code changes

 - Has Terraform scripts change

### Does this PR introduce a user-facing change?:
~~Several pods' and services' names are changed.
Eg. `tidb-cluster-grafana` -> `tidb-cluster-monitor-grafana`.
I'm not very sure whether I should change the code or change the document for that.~~
Already fixed

```release-note
NONE
```

### CODE REVIEW

File: charts/tidb-cluster/templates/monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                {{- range $kl, $vl := $valList }}
+                - {{ $vl }}
+                {{- end }}
+            {{- end }}
+  {{- end }}
+  {{- if .Values.monitor.tolerations }}
+  tolerations:
+{{ toYaml .Values.monitor.tolerations | indent 4 }}
+  {{- end }}
+{{- end }}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: Add the line in the end.

**Comment by Yisaer** [comment]
> PS. Release Notes Language Style Guide URL in the pr template is "not found" now, it should be Release Notes Language Style Guide (https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md)
May I create a pr for that later?

@longfangsong  Feel free to create any issue or request if there is any problem in our documents.

**Comment by Yisaer** [comment]
>Does this PR introduce a user-facing change?:
Several pods' and services' names are changed.
Eg. tidb-cluster-grafana -> tidb-cluster-monitor-grafana.
I'm not very sure whether I should change the code or change the document for that.

I think it is necessary to make sure when users migrate monitor into `TidbMonitor`, the origin components (at least the pvc ) wouldn't be changed or recreated if there isn't any new configuration.
I think an e2e case is necessary to ensure it.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+{{- if .Values.monitor.create }}
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  name: {{ template "cluster.name" . }}-monitor
+  namespace: {{ .Release.Namespace }}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: Please add the label information like other components defined in helm chart template.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      annotations: {}
+      portName: http-prometheus
+    resources:
+{{ toYaml .Values.monitor.prometheus.resources | indent 6 }}
+    baseImage: {{ (split ":" .Values.monitor.prometheus.image)._0 }}
+    version: {{ (split ":" .Values.monitor.prometheus.image)._1 }}
+    imagePullPolicy: {{ .Values.monitor.prometheus.imagePullPolicy }}
+    nodeSelector: {}
+    annotations: {}
+    tolerations: {}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: I think these configurations should be omitted.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  clusters:
+    - name: {{ template "cluster.name" . }}
+      namespace: {{ .Release.Namespace }}
+  persistent: {{ .Values.monitor.persistent }}
+  storage: {{ .Values.monitor.storage }}
+  prometheus:
+    logLevel: {{ .Values.monitor.prometheus.logLevel }}
+    service:
+      type: {{ .Values.monitor.prometheus.service.type }}
+      annotations: {}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: The service annotations should be exposed to users if they have their own configuraition.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    imagePullPolicy: {{ .Values.monitor.prometheus.imagePullPolicy }}
+    nodeSelector: {}
+    annotations: {}
+    tolerations: {}
+    reserveDays: {{ .Values.monitor.prometheus.reserveDays }}
+  {{- if .Values.monitor.reloader.create }}
+  reloader:
+    service:
+      type: {{ .Values.monitor.reloader.service.type }}
+      annotations: {}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: ditto.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    username: {{ .Values.monitor.grafana.username }}
+    password: {{ .Values.monitor.grafana.password }}
+    resources:
+{{ toYaml .Values.monitor.grafana.resources | indent 6 }}
+    baseImage: {{ (split ":" .Values.monitor.grafana.image)._0 }}
+    version: {{ (split ":" .Values.monitor.grafana.image)._1 }}
+    imagePullPolicy: {{ .Values.monitor.grafana.imagePullPolicy }}
+    service:
+      type: {{ .Values.monitor.grafana.service.type }}
+      annotations: {}
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: ditto.

File: charts/tidb-cluster/templates/monitor.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
+{{- if .Values.monitor.create }}
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  # for backward compatibility, name should not be {{ template "cluster.name" . }}-monitor
+  # or the names of the pods/svcs would be changed, like tidb-cluster-grafana -> tidb-cluster-monitor-grafana
```
Comment (Yisaer) [inline charts/tidb-cluster/templates/monitor.yaml]: For backward compatibility, I think this problem should be resolved in `TidbMonitor` Controller. I would create an new issue to solve this problem.

**Comment by Yisaer** [comment]
@aylei  Do you have any idea how to make new users to use monitor by `TidbMonitor` meanwhile the previous users could upgrade the `Operator` without losing data or cause a downtime.

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -36,7 +36,7 @@ Related changes
 If no, just leave the release note block below as is.
 If yes, a release note is required:
 Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/docs/release-note-guide.md) before writing the release note.
+Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.
```
Comment (cofyc) [inline .github/pull_request_template.md]: Can you open a separate PR for this? We can merge this first and don't mix two things together.

Comment (longfangsong) [reply .github/pull_request_template.md]: > Can you open a separate PR for this? We can merge this first and don't mix two things together.

I have created #1600 for this.

**Comment by Yisaer** [comment]
Currently, we had no idea to change the template for monitor with no downtime, so I have to close the issue.

**Pull request closed without merging**